src/arch.rs

```diff
@@ -57,14 +57,49 @@ impl<'a> JitValueWithInfoWrapper<'a> {
 impl<'a> AbiArgument for JitValueWithInfoWrapper<'a> {
     fn to_bytes(&self, buffer: &mut Vec<u8>) {
         match (self.value, self.info) {
-            (
-                value,
-                CoreTypeConcrete::Box(info)
-                | CoreTypeConcrete::NonZero(info)
-                | CoreTypeConcrete::Nullable(info)
-                | CoreTypeConcrete::Snapshot(info),
-            ) => {
-                // TODO: Allocate and use to_jit().
+            (value, CoreTypeConcrete::Box(info)) => {
+                let ptr = value
+                    .to_jit(self.arena, self.registry, self.type_id)
+                    .unwrap();
```

Comment:
Are these unwraps safe?
Should be implement an error handling here? Maybe we can make it in another PR if it is necesarry

Reply:
Ideally, yes. It could technically fail, but it'll most likely be because of an invalid Sierra program or internal Cairo error (the libraries we're using).
We can turn it into an error later on along with all the others once we fix our current bugs.